File: app/src/main/java/com/shen/xi/android/tut/WeiboImageViewActivity.java
package com.shen.xi.android.tut;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import com.google.api.client.json.JsonParser;
import com.google.inject.Inject;
import com.shen.xi.android.tut.weibo.WeiboClient;

import java.io.IOException;
import java.util.List;


public class WeiboImageViewActivity extends AbstractImageViewActivity {
    private static final String TAG = WeiboImageViewActivity.class.getName();
    private List<String> mStatusList;
    @Inject
    private WeiboClient weiboClient;

    public WeiboImageViewActivity() {
        super(R.menu.weibo_image_view);
        TuTModule.getInjector().injectMembers(this);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Create the adapter that will return a fragment for each of the three
        // primary sections of the activity.
        Intent intent = getIntent();
        int item = intent.getIntExtra(ITEM_POSITION, 0);
        String jsonList = intent.getStringExtra(JSON_LIST);

        try {
            JsonParser jsonParser = mJsonFactory.createJsonParser(jsonList);
            mStatusList = (List<String>) jsonParser.parseArray(List.class, String.class);
        } catch (IOException e) {
            e.printStackTrace();
            finish();
        }
        /*
          The {@link android.support.v4.view.PagerAdapter} that will provide
          fragments for each of the sections. We use a
          {@link FragmentPagerAdapter} derivative, which will keep every
          loaded fragment in memory. If this becomes too memory intensive, it
          may be best to switch to a
          {@link android.support.v4.app.FragmentStatePagerAdapter}.
        */
        ImagePagerAdapter pagerAdapter = new ImagePagerAdapter(
                getSupportFragmentManager(), mStatusList.size());
        setupPager(pagerAdapter, item);
    }

    @Override
    public String getImageUrlByOrder(int order) {
        return mStatusList.get(order);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        switch (id) {
            case R.id.menu_settings:
                return true;

            case android.R.id.home:
                this.finish();
                return true;

            default:
                return super.onOptionsItemSelected(item);
        }
    }

}

File: app/src/main/scala/com/shen/xi/android/tut/WeiboImageViewActivity.scala
package com.shen.xi.android.tut


import android.os.Bundle
import android.view.MenuItem
import com.google.inject.Inject
import com.shen.xi.android.tut.weibo.WeiboClient

object WeiboImageViewActivity {
  private val TAG = classOf[WeiboImageViewActivity].getName
}

class WeiboImageViewActivity extends AbstractImageViewActivity(R.menu.weibo_image_view) {

  import com.shen.xi.android.tut.AbstractImageViewActivity._
  import org.json4s._
  import org.json4s.native.JsonMethods._


  private var mStatusList: List[String] = null
  @Inject
  private var mWeiboClient: WeiboClient = null

  TuTModule.getInjector.injectMembers(this)

  override def onCreate(savedInstanceState: Bundle) = {
    super.onCreate(savedInstanceState)

    // Create the adapter that will return a fragment for each of the three
    // primary sections of the activity.
    val intent = getIntent
    val item = intent.getIntExtra(ITEM_POSITION, 0)
    val jsonList = intent.getStringExtra(JSON_LIST)

    mStatusList = for (JString(i) <- parse(jsonList)) yield i
    if (mStatusList.size == 0) {
      finish()
    } else {
      /** The 'android.support.v4.view.PagerAdapter' that will provide
        * fragments for each of the sections. We use a
        * 'FragmentPagerAdapter' derivative, which will keep every
        * loaded fragment in memory. If this becomes too memory intensive, it
        * may be best to switch to a
        * 'android.support.v4.app.FragmentStatePagerAdapter'.
        */
      val pagerAdapter = new ImagePagerAdapter(getSupportFragmentManager, mStatusList.size)
      setupPager(pagerAdapter, item)
    }
  }

  override def getImageUrlByOrder(order: Int) = mStatusList(order)

  // Handle action bar item clicks here. The action bar will
  // automatically handle clicks on the Home/Up button, so long
  // as you specify a parent activity in AndroidManifest.xml.
  override def onOptionsItemSelected(item: MenuItem) = item.getItemId match {
    case R.id.menu_settings => true

    case android.R.id.home =>
      this.finish()
      true

    case _ => super.onOptionsItemSelected(item)
  }
}
